src/core/thread/mle_router.cpp

```diff
@@ -3494,5 +3546,128 @@ ThreadError MleRouter::AppendPendingDataset(Message &aMessage)
     return error;
 }
 
+bool MleRouter::HasMinDowngradeNeighborRouters(void)
+{
+    return GetMinDowngradeNeighborRouters() >= kMinDowngradeNeighbors;
+}
+
+bool MleRouter::HasOneNeighborwithComparableConnectivity(const RouteTlv &aRoute, uint8_t aRouterId)
+{
+    uint8_t localLqi = 0;
+    uint8_t peerLqi = 0;
+    uint8_t routerCount = 0;
+
+    // process local neighbor routers
+    for (uint8_t i = 0; i < kMaxRouterId; i++)
+    {
+        if (i == mRouterId)
+        {
+            routerCount++;
+            continue;
+        }
+
+        // check if neighbor is valid
+        if (mRouters[i].mState == Neighbor::kStateValid)
+        {
+            // if peer is just neighbor
+            if (i == aRouterId)
```

Comment:
Here, the intend to skip the peer neighbor who is just the sender of MLE Advertisement, because in peer's Route64 TLV, both link_quality_in and link_quality_out to itself is 0. However, the link_quality_in/out from local router to this peer might not be 0. So skip local router itself and this peer router when checking link quality in Route64 TLV, only check if the rest neighbor routers can meet the second downgrade condition or not. Please correct me if this is not proper.